### PR TITLE
Fix: Addressing SonarQube issues

### PR DESCRIPTION
**Pull request by AdilOoze**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-1855

### Change description ###

Addressing SonarQube issues: adding null checks

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java [hunk 1]
```diff
@@ -250,7 +250,7 @@
             jurisdictionService.propagateJurisdictionIdsForSuperUserToCcd(professionalUser, userId);
             ResponseEntity<Object> responseEntity = createUserProfileFor(professionalUser, null, true,
                     false);
-            if (responseEntity.getStatusCode().is2xxSuccessful()) {
+            if (null != responseEntity.getBody() && responseEntity.getStatusCode().is2xxSuccessful()) {
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java:253]: better to check responseEntity.getStatusCode().is2xxSuccessful() first and then body

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java [hunk 2]
```diff
@@ -331,7 +331,7 @@
                 newUserCreationRequest.getEmail());
         ResponseEntity<Object> responseEntity = createUserProfileFor(professionalUser, roles, false,
                 false);
-        if (responseEntity.getStatusCode().is2xxSuccessful()) {
+        if (null != responseEntity.getBody() && responseEntity.getStatusCode().is2xxSuccessful()) {
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java:334]: same as above

**Review by shreedhar-hmcts — COMMENTED** [review]
any reason for deleting TelemetryRequestProcessor.java  ?

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java (hunk 2 above)
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java]: Addressed here: https://github.com/hmcts/rd-professional-api/pull/756/commits/269e0cdf95e711f02eabdfc64518cd59007083df

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java (hunk 1 above)
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java]: Addressed here: https://github.com/hmcts/rd-professional-api/pull/756/commits/269e0cdf95e711f02eabdfc64518cd59007083df

**Review by shreedhar-hmcts — APPROVED** [review]

**Review by shreedhar-hmcts — APPROVED** [review]

**Review by KOTLAPRA23 — APPROVED** [review]

**Pull request merged**